tools/mason/MasonPublish.chpl

```diff
@@ -242,32 +313,44 @@ proc getPackageName() throws {
     return name;
   }
   catch {
-    writeln('Error getting the name of your package, ensure your package is a mason project');
+    throw new owned MasonError('Issue getting the name of your package, ensure your package is a mason project.');
     exit(1);
   }
 }
 
 /* Adds package to the Bricks of the mason-registry branch and then adds the version.toml
  with the source url of the package's GitHub repo.
  */
-private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string) throws {
+private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string,registryPath : string, isLocal : bool) throws {
   try! {
-
     const toParse = open(projectLocal+ "/Mason.toml", iomode.r);
-    const url = gitUrl();
     var tomlFile = new owned(parseToml(toParse));
     const versionNum = tomlFile['brick']['version'].s;
-    mkdir(safeDir + "/mason-registry/Bricks/" + name);
-    const baseToml = tomlFile;
-    var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
-    var tomlWriter = newToml.writer();
-    baseToml["brick"].set("source", url[1..url.length-1]);
-    tomlWriter.write(baseToml);
-    tomlWriter.close();
+
+    if !isLocal {
+      mkdir(safeDir + "/mason-registry/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      const url = gitUrl();
+      baseToml["brick"].set("source", url[1..url.length-1]);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
     }
+    else {
+      mkdir(safeDir + "/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      baseToml["brick"].set("source", projectLocal);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
+      const gitMessageString = ('git tag -a v' + versionNum + ' -m  "' + name + '"');
+      gitC(projectLocal, gitMessageString);
+    }
+  }
   catch {
-    writeln('ERROR: ' + name + ' already exists in the Bricks');
-    rmTree(safeDir + '/');
-    exit(1);
+    if !isLocal then rmTree(safeDir + '/');
+    throw new owned MasonError('Unable to publish your package to the registry, make sure your package is a git repository.');
```

Comment:
So we replaced this but the case where it is actually already in the registry is now a bit confusing. 

This is the error a user sees when publishing the same package twice

> Unable to publish your package to the registry, make sure your package is a git repository.

Would it be possible to catch both when the repository is not a git repository and when the package had already been added to the registry?

Reply:
Better yet, an error message that tells them they need to update their package based on the presence of new commits compared to the old version!